cmd/tendermint/commands/compact.go

```diff
@@ -0,0 +1,70 @@
+package commands
+
+import (
+	"path/filepath"
+	"sync"
+
+	"github.com/spf13/cobra"
+	"github.com/syndtr/goleveldb/leveldb"
+	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/util"
+
+	"github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+func MakeCompactDBCommand(cfg *config.Config, logger log.Logger) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "experimental-compact-db",
+		Short: "force compacts the tendermint storage engine (only GoLevelDB supported)",
+		Long: `
+This is a temporary utility command that performs a force compaction on the state 
+and blockstores to reduce disk space for a pruning node. This should only be run 
+once the node has stopped. This command will likely be omitted in the future after
+the planned refactor to the storage engine.
+
+Currently, only GoLevelDB is supported.
+	`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if cfg.DBBackend != "goleveldb" {
+				logger.Info("compaction is currently only supported with goleveldb")
+				return
+			}
```

Comment:
I think maybe we should report an error here so the program will exit with a nonzero status (in case someone puts it in a script, they will need to know it isn't working).

```suggestion
		RunE: func(cmd *cobra.Command, args []string) error {
			if cfg.DBBackend != "goleveldb" {
				return errors.New("compaction is currently only supported with goleveldb")
			}
```

Reply:
Yeah that makes sense. I will switch this to returning an error